python/tutorials/gluon/08-warp-specialization.py

```diff
@@ -0,0 +1,655 @@
+"""
+Warp Specialization
+===================
+
+This tutorial covers warp specialization. In typical GPU kernels, all warps are
+executing cooperatively, meaning they perform parts of the same task. Warp
+specialization, however, is a technique where different warps in the kernel are
+doing completely different tasks.
+
+Warp specialization is typically used to overlap execution of different parts
+of the kernel. This is useful overlapping async operations with finer
+granularity than software pipelining, and we can overlap non-async operations
+that exercise different parts of the hardware without relying on precise
+SASS-level instruction interleaving.
```

Comment:
Isn't the advantage that the scheduling is dynamic due to the warp scheduler, rather than the granularity. This holds even for async ops, e.g. the mma can run ahead instead of waiting until the next load is ready to be issued.